docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,191 @@
+    - with-generated-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-account <account-id> beneficiary <beneficiary-account-id> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - list-keys <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - add-key <account-id>
+    - use-plaintext-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - generate-keypair network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-key <account-id> <public-key> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - propose-stake <account-id> <stake-amount-in-NEAR> <validation-node-public-key>
+    (maybe we should extract it into the validators extension)
+    - transaction signature options here (see below)
+
+  - TODO: "import account" (aka "login") commands
+```
+
+```
+contract
+  - call-view-function <account-id> <function-name> <function-args> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - call-change-function <account-id> <function-name> <function-args> --prepaid-gas <prepaid-gas> --attached-deposit <deposit-amount> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+  
+  - deploy <account-id> use-file <path-to-wasm-file>
+
+    - with-init-call <function-name> <function-args> --prepaid-gas <prepaid-gas> --attached-deposit <deposit-amount>
+      - transaction signature options here (see below)
+    - no-init-call
+      - transaction signature options here (see below)
+
+  - download <account-id> to-folder <path-to-download-folder> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - view-state <account-id> key-prefix <storage-key-prefix> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+```
+
+```
+tokens <owner-account-id>
+  - send-near <receiver-account-id> <amount-in-NEAR> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - send-ft <ft-contract-account-id> <receiver-account-id> <amount-in-fungible-tokens> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+  
+  - send-nft <nft-contract-account-id> <receiver-account-id> <token-id> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+  
+  - view-near-balance network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+  
+  - view-ft-balance <ft-contract-account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+  
+  - view-nft-list <nft-contract-account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
```

Comment:
Not sure what this command will return.

Reply:
It's my understanding that this should return a bunch of `JsonToken` information such as metadata, owner, token ID etc...

To that point, however, should we include from index, limit etc.. into the parameters of this call?

Reply:
@BenKurrek I would expect CLI would handle pagination internally and query all of the owned NFTs even if it takes multiple function calls.

Reply:
Makes sense - what if the user didn't want all that info returned? What if they just wanted to see everything past a starting index of 50 for example?

Reply:
@BenKurrek This is a good point, but I think we can discuss it in more details later. We will need to have more features built into it like filtering and history of NFT operations. If there are <100 NFTs, we don't need to solve the pagination issue and can just fetch all of them, but if there are more than 100 NFTs, it makes no sense to pull all of those anyway, and manually scanning through the pages is not fan either, so we need to identify the use-cases and solve those instead of asking users to paginate though hundreds of NFTs manually.

Reply:
Yea for sure - perhaps we can add this to the NFT extension and talk about it when we implement that.